internal/config/file_source.go

```diff
@@ -0,0 +1,133 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/milvus-io/milvus/internal/log"
+	"github.com/spf13/cast"
+	"github.com/spf13/viper"
+	"go.uber.org/zap"
+)
+
+type FileSource struct {
+	sync.RWMutex
+	file    string
+	configs map[string]string
+}
+
+func NewFileSource(file string) *FileSource {
+	fs := &FileSource{file: file, configs: make(map[string]string)}
+	fs.loadFromFile(file)
+	log.Info("Init File Source")
+	for key, value := range fs.configs {
+		log.Debug("Got", zap.String("key", key), zap.String("value", value))
+	}
+	return fs
+}
+
+// GetConfigurationByKey implements ConfigSource
+func (fs *FileSource) GetConfigurationByKey(key string) (string, error) {
+	v, ok := fs.configs[key]
+	if !ok {
+		return "", fmt.Errorf("key not found: %s", key)
+	}
+	return v, nil
+}
+
+// GetConfigurations implements ConfigSource
+func (fs *FileSource) GetConfigurations() (map[string]string, error) {
+	configMap := make(map[string]string)
+	fs.Lock()
+	defer fs.Unlock()
+	for k, v := range fs.configs {
+		configMap[k] = v
+	}
+	return configMap, nil
+}
+
+// GetPriority implements ConfigSource
+func (fs *FileSource) GetPriority() int {
+	return 3
```

Comment:
will it be better to have one place to define all kind of source 's priority?

Reply:
Agree. It would be clearer if new source is added and the priority will not be messed up.

Reply:
Getting priority by the method is more extendable 🤔